it/src/main/java/com/google/cloud/teleport/it/bigtable/DefaultBigtableResourceManager.java

```diff
@@ -0,0 +1,421 @@
+    try (BigtableDataClient dataClient =
+        bigtableResourceManagerClientHandler.getBigtableDataClient()) {
+      for (RowMutation tableRow : tableRows) {
+        dataClient.mutateRow(tableRow);
+      }
+    } catch (Exception e) {
+      throw new BigtableResourceManagerException("Failed to write mutations.", e);
+    }
+
+    LOG.info("Successfully sent mutations to instance {}.", instanceId);
+  }
+
+  @Override
+  public synchronized ImmutableList<Row> readTable(String tableId) {
+    checkHasInstance();
+    checkHasTable(tableId);
+
+    // List to store fetched rows
+    ImmutableList.Builder<Row> tableRowsBuilder = ImmutableList.builder();
+
+    LOG.info("Reading all rows from {}.{}", instanceId, tableId);
+
+    // Fetch the Bigtable data client and read all the rows from the table given by tableId
+    try (BigtableDataClient dataClient =
+        bigtableResourceManagerClientHandler.getBigtableDataClient()) {
+
+      Query query = Query.create(tableId);
+      ServerStream<Row> rowStream = dataClient.readRows(query);
+      for (Row row : rowStream) {
+        tableRowsBuilder.add(row);
+      }
+
+    } catch (IOException | NotFoundException e) {
+      throw new BigtableResourceManagerException("Error occurred while reading table rows.", e);
+    }
+
+    ImmutableList<Row> tableRows = tableRowsBuilder.build();
+    LOG.info("Loaded {} rows from {}.{}", tableRows.size(), instanceId, tableId);
+
+    return tableRows;
+  }
+
+  @Override
+  public synchronized void cleanupAll() {
+    LOG.info("Attempting to cleanup manager.");
+    try {
+      bigtableResourceManagerClientHandler
+          .getBigtableInstanceAdminClient()
+          .deleteInstance(instanceId);
+      bigtableResourceManagerClientHandler.getBigtableInstanceAdminClient().close();
+      bigtableResourceManagerClientHandler.getBigtableTableAdminClient().close();
+      hasInstance = false;
+    } catch (Exception e) {
+      throw new BigtableResourceManagerException("Failed to delete resources.", e);
+    }
+
+    LOG.info("Manager successfully cleaned up.");
+  }
+
+  /** Builder for {@link DefaultBigtableResourceManager}. */
+  public static final class Builder {
+
+    private final String testId;
+    private final String projectId;
+    private String instanceId;
+    private CredentialsProvider credentialsProvider;
+    private final BigtableInstanceAdminSettings.Builder bigtableInstanceAdminSettings;
+    private final BigtableTableAdminSettings.Builder bigtableTableAdminSettings;
+    private final BigtableDataSettings.Builder bigtableDataSettings;
+
+    private Builder(String testId, String projectId) {
+
+      // Generate and format ID's
+      if (testId.length() > MAX_TEST_ID_LENGTH) {
+        testId = generateNewId(testId, MAX_TEST_ID_LENGTH);
+      }
+      this.testId = testId;
+      this.projectId = projectId;
+      this.instanceId = generateInstanceId(testId);
```

Comment:
I think it's better to to generate instance ID and AdminSettings in the resource manager constructor.

Reply:
I'm not sure how to accomplish this since the Client Settings are built before being wrapped in the BigtableResourceManagerClientHandler class, and the ID's must be set in each of the Settings objects before building.

Reply:
I'd move `generateInstanceId(testId)` to the `build()` method then, to generate a new id for every new RM instance constructed by the builder.

Reply:
Fixed.